### PR TITLE
feat: Add support for GitHub Enterprise authentication

### PR DESCRIPTION
**Pull request by kylecarbs**
This was manually tested with GitHub Enterprise v3.6.0-rc1.

### CODE REVIEW

**Review by ammario — APPROVED** [review]
Docs LGTM

**Review by bpmct — APPROVED** [review]
Looks sensible to me

File: cli/server_test.go [hunk 1]
```diff
@@ -429,6 +429,39 @@ func TestServer(t *testing.T) {
 		cancelFunc()
 		<-serverErr
 	})
+	t.Run("GitHubOAuth", func(t *testing.T) {
```
Comment (bpmct) [inline cli/server_test.go:432]: so this is an entirely new test for all GitHub OAuth?

**Review by coadler — APPROVED** [review]

File: cli/server_test.go (hunk 1 above)
Comment (kylecarbs) [reply cli/server_test.go:432]: Yup! The OAuth callback code itself is tested quite extensively in the other handler, so this just ensures it enables properly, and does a correct callback URL for Enterprise.

**Comment by OlgaKhmelevskaya** [comment]
Hello @kylecarbs , could you please add reference to the issue this pull request is related to?

**Pull request merged**